src/main/java/com/github/fge/jsonpatch/JsonPatchDeserializer.java

```diff
@@ -0,0 +1,36 @@
+package com.github.fge.jsonpatch;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+
+import java.io.IOException;
+
+/**
+ * JsonPatchDeserializer allows you to register a custom JsonPatchFactory for jackson to use
+ * during deserialization.
+ */
+public class JsonPatchDeserializer
+    extends JsonDeserializer<JsonPatch>
+{
+    private JsonPatchFactory factory;
+    public JsonPatchDeserializer(JsonPatchFactory factory)
+    {
+        this.factory = factory;
+    }
+
+    @Override
+    public JsonPatch deserialize(final JsonParser jp,
+        final DeserializationContext ctxt)
+        throws IOException, JsonProcessingException
+    {
+        final JsonNode node = jp.readValueAsTree();
```

Comment:
What do you think about using mix-in annotations to achieve custom deserialization into our extended JSON-Patch operations? We'd mix in JsonSubTypes annotations that list all of our extended JSON-Patch operations.  With this, we shouldn't even need the JsonPatchOperationFactory anymore! http://wiki.fasterxml.com/JacksonMixInAnnotations.